src/WebJobs.Extensions.CosmosDB/Trigger/CosmosDBTriggerAttributeBindingProvider.cs

```diff
@@ -169,52 +164,67 @@ internal static TimeSpan ResolveTimeSpanFromMilliseconds(string nameOfProperty,
 
         private string ResolveAttributeConnectionString(CosmosDBTriggerAttribute attribute)
         {
-            string connectionString = this._monitorConnectionString;
-            if (!TryReadFromSettings(attribute.ConnectionStringSetting, connectionString, out connectionString))
+            string connectionString = ResolveConnectionString(attribute.ConnectionStringSetting, nameof(CosmosDBTriggerAttribute.ConnectionStringSetting));
+
+            if (string.IsNullOrEmpty(connectionString))
             {
+                string attributeProperty = $"{nameof(CosmosDBTriggerAttribute)}.{nameof(CosmosDBTriggerAttribute.ConnectionStringSetting)}";
+                string optionsProperty = $"{nameof(CosmosDBOptions)}.{nameof(CosmosDBOptions.ConnectionString)}";
                 throw new InvalidOperationException(
-                    string.Format(CultureInfo.CurrentCulture,
-                    "The CosmosDBTriggerAttribute.ConnectionStringSetting '{0}' property specified does not exist in the Application Settings.",
-                    attribute.ConnectionStringSetting));
+                    $"The CosmosDBTrigger connection string must be set either via a '{Constants.DefaultConnectionStringName}' IConfiguration connection string, via the {attributeProperty} property or via {optionsProperty}.");
```

Comment:
It looks like this complicated error message formatting is duplicated? Would be good to centralize it on one place.

Reply:
Yup, will do.